### ISSUE TITLE
Allow 1 instance option

### ISSUE THREAD

**Issue by mu-us61** [giuspen/cherrytree#2150]
**Is your feature request related to a problem? Please describe.**
as it is when i click cherrytree to open it opens a new instance with empty project, 
I do use only 1 project and want it open even when i click multiple times to cherrytree

**Describe the solution you'd like**
a new option on settings to allow cherrytree to use only 1 instance, so no new empty cherrytree bug I guess

**Comment by gitnard** [comment]
Here is how I solved your issue (both initially and later). (My OS: Debian and MX-Linux, with Xfce.)

1) If you only want to open only one Cherrytree document (from the OS default menu for opening applications), then when in your document, at Cherrytree  >File  >Preferences  >Miscellaneous, Miscellaneous, YES Reload Document From Last Session.

2) If you want to efficiently open one (or more) Cherrytree document(s), then create keyboard application shortcut(s) to start Cherrytree with your document opened.

See Cherrytree  >Help  >Online Manual F1  >5.6. Command Line.  "Cherrytree supports command line argument for starting the application via command line.  Positional arguments:  filepath  Defines the document to open in Cherrytree. Replace filepath with the path of a file."

Here are my four keyboard shortcuts, showing command with filepath/filename.ctb, and showing keyboard shortcut.
```
cherrytree /home/gn/1WORK/CT/gnLinux.ctb       Ctrl+Alt+L
cherrytree /home/gn/1WORK/CT/gnNotes.ctb       Ctrl+Alt+N
cherrytree /home/gn/1WORK/CT/gnQuadr.ctb       Ctrl+Alt+Q
cherrytree /home/gn/1WORK/CT/gnZscratch.ctb    Ctrl+Alt+Z
```
Each keyboard shortcut efficiently opens a separate filename.ctb. Intentionally or accidentally reentering the keyboard shortcut for an already open filename.ctb will raise (make visible if below other windows) that single Cherrytree instance (and not raise other Cherrytree instances that might also be open). As well, the OS default menu for opening applications remains available for new Cherrytree startup.

**Comment by mu-us61** [comment]
yes my settings is already setted for Reload Document From Last Session , but when opening a new instance it does not work if you have already have one open , and why I do that because sometimes it is easy for me to click desktop app instead of  using tap or clicking system tray icon

how to make this happen :
1- just open  cherrytree with a document from last session 
2- open cherrytree again without closing the first one 
3- new cherrytree instance opens with empty session , ( first one stays )

what should have happened:
 it should open same cherrytree instance with loaded default document ( i use only 1 )

my OS is , Debian 11 KDE 
cherrytree version is 0.99.51

![tree](https://user-images.githubusercontent.com/73366454/202869079-f9bb8078-ad8a-4c69-a1ef-bb17cdb3707c.gif)

**Comment by gitnard** [comment]
Did you make a keyboard shortcut to your .ctb file?

Does initial use of the shortcut open a Cherrytree session showing your .ctb?

Does subsequent use of the shortcut raise your Cherrytree session when it was hidden below other windows?

**Comment by mu-us61** [comment]
> Did you make a keyboard shortcut to your .ctb file?
> 
> Does initial use of the shortcut open a Cherrytree session showing your .ctb?
> 
> Does subsequent use of the shortcut raise your Cherrytree session when it was hidden below other windows?

I was using flatpak then switch to snap to try if gonna work but result was the same 
then as a workaround I did as you say and it worked thanks for help, I edit the desktop icon file and added .ctb's path , now it only opens the same instance and only 1 instance

`Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/cherrytree_cherrytree.desktop /snap/bin/cherrytree --new_window /home/muhammet/Documents/DriveBackup/CherryTree/21_12_newone.ctb %f`

**Comment by onchov** [comment]
I second the original feature request to include an option to only allow one instance of the app, also as a solution to an annoyance I'm experiencing, where I 'wake up' to having a couple of empty instances open out of nowhere on top of the only one I normally use.

But even without that bug (I reckon), it's only sensible to have such an option in this kind of app.
